tico/passes/remove_redundant_reshape.py

```diff
@@ -295,10 +295,13 @@ def call(self, exported_program: ExportedProgram) -> PassResult:
             # Check condition
```

Comment:
Could you add this test case?

reshape_2_input_shape = [1,1,6]
reshape_3_input_shape = [1,1,6]
reshape_2_shape = [1,2,3]
reshape_3_shape = [1,2,3]
softmax_shape = [1,2,3]
reshape_1_shape = [1,1,6]

I'm not sure the current code can block the case.

(optional) Code is very difficult to read. It would be better to separate the function to two parts.
1) Get arguments.
2) Check conditions (add comment for each condition).

Reply:
@jinevening 

> I'm not sure the current code can block the case.

Is it the case that should be blocked? 

> Code is a little bit difficult to read

I guess it would be better to do this in refactoring PR.

Reply:
Ah, now I got it! I'll add a condition for this. Thanks:)